# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [17]:
# "tobs" is "temperature observations"
hawaii_df = pd.read_csv('Resources/hawaii_measurements.csv')
hawaii_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [18]:
# Convert the date column format from string to datetime
# convert the 'Date' column to datetime format
hawaii_df.date = pd.to_datetime(hawaii_df.date, infer_datetime_format=True)

In [19]:
# Set the date column as the DataFrame index
hawaii_df = hawaii_df.set_index(hawaii_df['date'])
hawaii_df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [20]:
# Drop the date column
hawaii_df = hawaii_df.drop(columns='date')
hawaii_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [21]:
from scipy import stats

In [22]:
# Filter data for desired months
juneDate = hawaii_df[hawaii_df.index.month == 6]
juneDate.head()

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77

In [23]:
decemberDate = hawaii_df[hawaii_df.index.month == 12]
decemberDate.head()

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64

In [30]:
# Identify the average temperature for June
juneDate.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [32]:
# Identify the average temperature for December
decemberDate.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [31]:
decemberDate.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [33]:
# Create collections of temperature data
juneTemp = juneDate.tobs
juneTemp

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [34]:
decemberTemp = decemberDate.tobs
decemberTemp

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [36]:
# Run paired t-test
stats.ttest_ind(juneTemp,decemberTemp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The mean temperature difference between the month of June and the month of December is 3.9 degrees Fahrenheit. The low p-value of the two t-values indicates that the difference is statistically significant to the sampled sets, and in all likelihood did not occur by chance. Now with that said, in the context of what this is testing, peak summer month and peak winter month, the actual difference of 3.9 degrees does not equate to a huge swing in temperature differentials, hence why Hawaii and tropical climates in general tend to be considered 'mild all year' sort of climates.